components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/util/ActionManagementAuditLoggerTest.java

```diff
@@ -0,0 +1,268 @@
+     * @return Value of the extracted field.
+     * @throws NoSuchFieldException   if the provided field does not exist.
+     * @throws IllegalAccessException if the provided field is not accessible.
+     */
+    private String extractField(String fieldName, AuditLog.AuditLogBuilder auditLogBuilder)
+            throws NoSuchFieldException, IllegalAccessException {
+
+        Field dataField = AuditLog.AuditLogBuilder.class.getDeclaredField(fieldName);
+        dataField.setAccessible(true);
+        return (String) dataField.get(auditLogBuilder);
+    }
+
+    /**
+     * Assert data fields related to the Action object of the captured audit logger.
+     *
+     * @param auditLogBuilder {@link AuditLog.AuditLogBuilder} instance.
+     * @throws NoSuchFieldException   if the provided field does not exist.
+     * @throws IllegalAccessException if the provided field is not accessible.
+     */
+    private void assertActionData(AuditLog.AuditLogBuilder auditLogBuilder)
+            throws NoSuchFieldException, IllegalAccessException {
+
+        Assert.assertEquals(extractMapByField("ActionId", auditLogBuilder), "action-test-id");
+        Assert.assertEquals(extractMapByField("ActionName", auditLogBuilder), "Test Action");
+        Assert.assertEquals(extractMapByField("ActionType", auditLogBuilder),
+                Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getActionType());
+        Assert.assertEquals(extractMapByField("ActionStatus", auditLogBuilder), Action.Status.ACTIVE.value());
+        Assert.assertEquals(extractMapByField("ActionDescription", auditLogBuilder),
+                "This is a test action.");
+        Assert.assertEquals(extractEndpointMapByField("AuthenticationScheme", auditLogBuilder),
+                Authentication.Type.BEARER.getName());
+        Assert.assertEquals(extractEndpointMapByField("EndpointUri", auditLogBuilder),
+                "https://test.com");
+    }
+
+    /**
+     * Assert generic data fields in audit logger.
+     *
+     * @param auditLogBuilder {@link AuditLog.AuditLogBuilder} instance.
+     * @param operation       Operation to be logged.
+     * @throws NoSuchFieldException   if the provided field does not exist.
+     * @throws IllegalAccessException if the provided field is not accessible.
+     */
+    private void assertAuditLoggerData(AuditLog.AuditLogBuilder auditLogBuilder,
+                                       ActionManagementAuditLogger.Operation operation)
+            throws NoSuchFieldException, IllegalAccessException {
+
+        Assert.assertEquals(extractField("initiatorId", auditLogBuilder), "initiator-id-test");
+        Assert.assertEquals(extractField("targetId", auditLogBuilder), "System");
+        Assert.assertEquals(extractField("targetType", auditLogBuilder), "Action");
+        switch (operation) {
+            case ADD:
+                Assert.assertEquals(extractField("action", auditLogBuilder), "add-action");
```

Comment:
```suggestion
                Assert.assertEquals(extractField("action", auditLogBuilder), "add-action");
```
Shall we use the defined constants for this.
Eg, `ActionManagementAuditLogger.Operation.ADD`

Reply:
This is updated with 40e761402ef38df6e482aa8d3149338e2da602ca